sprints/dashboard/tasks.py

```diff
@@ -84,88 +97,81 @@ def complete_sprints() -> None:
+                state='active',
             )
-            issue_keys = [issue.key for issue in issues]
-
-            # It is not mentioned in Python lib docs, but the limit for the issue-moving queries is 50 issues. Source:
-            # https://developer.atlassian.com/cloud/jira/software/rest/#api-rest-agile-1-0-backlog-issue-post
-            # https://developer.atlassian.com/cloud/jira/software/rest/#api-rest-agile-1-0-sprint-sprintId-issue-post
-            batch_size = 50
-            if not settings.DEBUG:  # We really don't want to trigger this in the dev environment.
-                # Remove archived tickets from the active sprint. Leaving them might interrupt closing the sprint.
-                for i in range(0, len(archived_issue_keys), batch_size):
-                    batch = archived_issue_keys[i:i + batch_size]
-                    conn.move_to_backlog(batch)
-
-                # Close the active sprint.
-                if counter == len(cells) - 1:  # TODO: Remove this after ending the current sprint.
-                    conn.update_sprint(
-                        active_sprint.id,
-                        name=active_sprint.name,
-                        startDate=active_sprint.startDate,
-                        endDate=active_sprint.endDate,
-                        state='closed',
-                    )
-
-                # Move issues to the next sprint from the closed one.
-                for i in range(0, len(issue_keys), batch_size):
-                    batch = issue_keys[i:i + batch_size]
-                    conn.add_issues_to_sprint(next_sprint.id, batch)
-
-                # Open the next sprint.
-                start_date = datetime.now()
-                end_date = datetime.now() + timedelta(days=settings.SPRINT_DURATION_DAYS)
-                conn.update_sprint(
-                    next_sprint.id,
-                    name=next_sprint.name,
-                    startDate=start_date.isoformat(),
-                    endDate=end_date.isoformat(),
-                    state='active',
-                )
-
-                # Ensure that the next sprint exists. If it doesn't exist, create it.
-                future_next_sprint_number = get_sprint_number(next_sprint) + 1
-                future_name_date = (end_date + timedelta(days=1)).strftime('%Y-%m-%d')
-                future_end_date = end_date + timedelta(days=settings.SPRINT_DURATION_DAYS)
-                conn.create_sprint(
-                    name=f'{cell.key}.{future_next_sprint_number} ({future_name_date})',
-                    board_id=cell.board_id,
-                    startDate=end_date.isoformat(),
-                    endDate=future_end_date.isoformat(),
-                )
+
+            # Ensure that the next sprint exists. If it doesn't exist, create it.
+            future_next_sprint = get_next_sprint(sprints, next_sprint)
+            if not future_next_sprint:
+                create_next_sprint_task.delay(board_id)
```

Comment:
That's a great idea!
I'll leave this for later, though, because `get_sprint_end_date` calls `get_next_sprint` and it doesn't have the Jira connection arg, so we would need to change a couple things to get this working.